zebra-state/src/request.rs

```diff
@@ -234,6 +235,18 @@ pub struct ContextuallyVerifiedBlock {
     pub(crate) chain_value_pool_change: ValueBalance<NegativeAllowed>,
 }
 
+/// Data related to an invalidated block including the [`ContextuallyVerifiedBlock`] and
+/// descendants.
+#[derive(Clone, Debug)]
+pub struct InvalidatedBlockData {
+    /// The block that was invalidated.
+    pub block: ContextuallyVerifiedBlock,
+    /// All descendant blocks that were also removed in order by [`block::Height`].
+    pub descendants: BTreeMap<block::Height, ContextuallyVerifiedBlock>,
```

Comment:
There's a suggestion below for removing this struct altogether, but if we're keeping it, minor documentation update:

```suggestion
/// [`ContextuallyVerifiedBlock`]s that were invalidated via the `invalidate` RPC method
/// and removed from the non-finalized state, as well as any of their descendants that may
/// have also been removed.
#[derive(Clone, Debug)]
pub struct InvalidatedBlockData {
    /// The block that was invalidated.
    pub block: ContextuallyVerifiedBlock,
    /// All child blocks of the invalidated block that were removed in order by [`block::Height`].
    pub descendants: BTreeMap<block::Height, ContextuallyVerifiedBlock>,
```